client/my-sites/stats/stats-period-navigation/index.jsx

```diff
@@ -243,7 +243,7 @@ class StatsPeriodNavigation extends PureComponent {
 		const path = `/stats/${ period }/${ slug }`;
 		const url = getPathWithUpdatedQueryString( { chartStart, chartEnd }, path );
 
-		page( url );
+		page.redirect( url );
```

Comment:
I don't know whether we should do this.

- User might want actually go back to the previous date selection, which could be achieved by clicking the back button in the browser. However after this change and https://github.com/Automattic/wp-calypso/pull/101003,  it wouldn't be possible. We might not want to break the ability to navigate from the browser.
- User might move out of the date range. For example, if the date range is Last 7 Day, user selected the first day of the week, and then click back, users shouldn't be able to go up again, because they are out of the previously selected range, and it could be confusing.

Reply:
I was considering allowing users to return to the upper layer even after navigating with arrows. However, keeping the history would indeed benefit users who want to visit all previous navigations by the browser's last page button. I will try another approach to fix this. 👍🏼

Reply:
Updated in [19a38cf](https://github.com/Automattic/wp-calypso/pull/101239/commits/19a38cfa29a755e3891c68126b853a2a6319c405) with another approach.